### PR TITLE
fix: missing dependency for Jazzy

### PR DESCRIPTION
**Pull request by f0reachARR**
## Description

This PR add / modify dependencies written in `package.xml`.
The changed dependencies are almost about `ament_index_cpp`.
For reasons unknown, Jazzy has a build error due to not having a dependency on ament_index_cpp.
We have found similar errors in other ROS packages and have confirmed that they are not specific to the author's environment.

Reference: #7598 

## Tests performed

`colcon test` passed.
This PR does not change the behavior of Autoware.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/autoware_static_centerline_generator/package.xml [hunk 1]
```diff
@@ -29,6 +29,7 @@
   <depend>geography_utils</depend>
   <depend>geometry_msgs</depend>
   <depend>global_parameter_loader</depend>
+  <depend>grid_map_ros</depend>
```
Comment (mitsudome-r) [inline planning/autoware_static_centerline_generator/package.xml:32]: Could you tell me the error output when if don't have grid_map_ros added into this dependency?
I have quickly searched the code with `grid_map` as a keyword in autoware_static_centerline_generator package, but I couldn't get any search results.

Comment (f0reachARR) [reply planning/autoware_static_centerline_generator/package.xml:32]: I have some errors without `grid_map_ros`.

```
<command-line>: fatal error: grid_map_core/eigen_plugins/FunctorsPlugin.hpp: No such file or directory
compilation terminated.
gmake[2]: *** [CMakeFiles/main.dir/build.make:132: CMakeFiles/main.dir/src/utils.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
<command-line>: fatal error: grid_map_core/eigen_plugins/FunctorsPlugin.hpp: No such file or directory
compilation terminated.
gmake[2]: *** [CMakeFiles/main.dir/build.make:118: CMakeFiles/main.dir/src/centerline_source/bag_ego_trajectory_based_centerline.cpp.o] Error 1
<command-line>: fatal error: grid_map_core/eigen_plugins/FunctorsPlugin.hpp: No such file or directory
compilation terminated.
gmake[2]: *** [CMakeFiles/main.dir/build.make:104: CMakeFiles/main.dir/src/centerline_source/optimization_trajectory_based_centerline.cpp.o] Error 1
<command-line>: fatal error: grid_map_core/eigen_plugins/FunctorsPlugin.hpp: No such file or directory
compilation terminated.
gmake[2]: *** [CMakeFiles/main.dir/build.make:90: CMakeFiles/main.dir/src/static_centerline_generator_node.cpp.o] Error 1
<command-line>: fatal error: grid_map_core/eigen_plugins/FunctorsPlugin.hpp: No such file or directory
compilation terminated.
gmake[2]: *** [CMakeFiles/main.dir/build.make:76: CMakeFiles/main.dir/src/main.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:625: CMakeFiles/main.dir/all] Error 2
```

This is caused by defining `EIGEN_DENSEBASE_PLUGIN` by `grid_map_ros`.
`EIGEN_DENSEBASE_PLUGIN` affects [Eigen behavior](https://eigen.tuxfamily.org/dox/TopicPreprocessorDirectives.html), and caues build error.
`grid_map_ros` is an indirect dependency of `static_centerline_generator`, but does not appear in include directory of `static_centerline_generator`.

**Comment by SakodaShintaro** [comment]
Thank you for your pull request.
For the localization component, it seems to be fine, but there is a conflict in localization/pose_instability_detector/package.xml.
Also, if you submit pull requests for each component separately, it may be easier to review and merge them.

**Comment by Shin-kyoto** [comment]
Thank you for your PR!
Can you submit pull requests for each component separately, please?

**Comment by Shin-kyoto** [comment]
@f0reachARR 
Thank you for creating the PR.
Would it be possible for you to resolve the conflicts? 

To keep Autoware healthy, we close PRs if there is no response for more than a week after requesting conflict resolution. If this PR is still active, please let us know. Even if this PR is closed, you are welcome to reopen it once the conflicts have been resolved.

**Comment by HansRobo** [comment]
@Shin-kyoto Thank you for friendly ping!
This PR is still alive but the auther is taking some time off.
So, you have any requests or comments, please mention me.

The status of this PR is currently under investigation for an unclear point discussed [here](https://github.com/autowarefoundation/autoware.universe/pull/7600#discussion_r1647735004).

To make this status easier to understand, I have marked the PR as draft.

**Comment by Shin-kyoto** [comment]
> @Shin-kyoto Thank you for friendly ping! This PR is still alive but the auther is taking some time off. So, you have any requests or comments, please mention me.
> 
> The status of this PR is currently under investigation for an unclear point discussed [here](https://github.com/autowarefoundation/autoware.universe/pull/7600#discussion_r1647735004).
> 
> To make this status easier to understand, I have marked the PR as draft.

Thank you very much for telling the situation of this PR!!
We understand the situation :+1:

**Comment by f0reachARR** [comment]
Sorry for late reply.
Now I can resolve conflicts, however it requires many approval to merge this PR.
I would like to close this PR, and split this PR by each components (like sensing, perception). Is this OK?

**Pull request closed without merging**